### PR TITLE
feat: team repository

### PR DESCRIPTION
**Pull request by nunocaseiro**
## Proposed Changes

  - Add team and team-user repository

### CODE REVIEW

File: backend/src/libs/repositories/mongo/mongo-generic.repository.ts [hunk 1]
```diff
@@ -1,32 +1,47 @@
...
+
+	findAll(
+		selectedValues?: SelectedValues<T>,
+		populate?: PopulateOptions | (PopulateOptions | string)[]
+	): Promise<T[]> {
+		return this._repository.find().select(selectedValues).populate(populate).exec();
 	}
 
-	get(id: any, selectedValues?: SelectedValues<T>): Promise<T> {
+	findOneById(id: any, selectedValues?: SelectedValues<T>, populate?: any): Promise<T> {
```
Comment (rpvsilva) [inline backend/src/libs/repositories/mongo/mongo-generic.repository.ts:24]: You should type the `populate` property

File: backend/src/modules/teams/services/create.team.service.ts [hunk 2]
```diff
@@ -1,44 +1,48 @@
...
 	constructor(
-		@InjectModel(Team.name) private teamModel: Model<TeamDocument>,
-		@InjectModel(TeamUser.name) private teamUserModel: Model<TeamUserDocument>
+		@Inject(TYPES.repositories.TeamRepository)
+		private readonly teamRepository: TeamRepositoryInterface,
+		@Inject(TYPES.repositories.TeamUserRepository)
+		private readonly teamUserRepository: TeamUserRepositoryInterface
 	) {}
 
 	async createTeamUsers(teamUsers: TeamUserDto[], teamId: string) {
```
Comment (rpvsilva) [inline backend/src/modules/teams/services/create.team.service.ts:20]:
```suggestion
	createTeamUsers(teamUsers: TeamUserDto[], teamId: string) {
```

**Review by rpvsilva — CHANGES_REQUESTED** [review]

File: backend/src/libs/repositories/interfaces/base.repository.interface.ts [hunk 3]
```diff
@@ -2,17 +2,31 @@ import { UpdateQuery } from 'mongoose';
...
+
+	findOneAndRemoveByField(fields: ModelProps<T>, withSession: boolean): Promise<T>;
+
+	startTransaction();
+
+	commitTransaction();
+
+	abortTransaction();
+
+	endSession();
```
[the remark this answers is not in the document]
Comment (nunocaseiro) [reply backend/src/libs/repositories/interfaces/base.repository.interface.ts:31]: These types just return void, do you think they should have the void type?

[the remark this answers is not in the document]
Comment (rpvsilva) [reply backend/src/libs/repositories/interfaces/base.repository.interface.ts:31]: Yes please, since it's an interface it's important to be fully typed

**Review by rpvsilva — APPROVED** [review]
LGTM 🔥

**Pull request merged**